### ISSUE TITLE
apf_controller.go:411] "Unable to derive new concurrency limits" err="impossible: ran out of bounds to consider in bound-constrained problem" 

### ISSUE THREAD

**Issue by ming12713** [kubernetes/kubernetes#120032]
### What happened?

hi, we have 3 master nodes , one of nodes errors output this err

kubectl -n kube-system logs kube-apiserver-loshu-k8s-master02
```
E0818 02:26:14.906899       1 apf_controller.go:411] "Unable to derive new concurrency limits" err="impossible: ran out of bounds to consider in bound-constrained problem" plNames=[catch-all global-default leader-election node-high system workload-high workload-low] items=[{target:NaN lowerBound:13 upperBound:613} {target:24 lowerBound:24 upperBound:649} {target:25 lowerBound:25 upperBound:625} {target:73 lowerBound:73 upperBound:698} {target:50 lowerBound:50 upperBound:674} {target:49 lowerBound:49 upperBound:698} {target:24 lowerBound:24 upperBound:845}]
E0818 02:26:24.907665       1 apf_controller.go:411] "Unable to derive new concurrency limits" err="impossible: ran out of bounds to consider in bound-constrained problem" plNames=[global-default leader-election node-high system workload-high workload-low catch-all] items=[{target:24 lowerBound:24 upperBound:649} {target:25 lowerBound:25 upperBound:625} {target:73 lowerBound:73 upperBound:698} {target:50 lowerBound:50 upperBound:674} {target:49 lowerBound:49 upperBound:698} {target:24 lowerBound:24 upperBound:845} {target:NaN lowerBound:13 upperBound:613}]
I0818 02:26:30.011746       1 trace.go:219] Trace[1755777139]: "Get" accept:application/json,audit-id:28780c1b-d603-4457-85b8-08fcde26f78c,client:10.0.162.31,protocol:HTTP/1.1,resource:pods,scope:resource,url:/api/v1/namespaces/workflow/pods/navi-quality-union-pod-g7mmd5u0/log,user-agent:OpenAPI-Generator/23.6.0/python,verb:GET (18-Aug-2023 02:25:52.635) (total time: 37376ms):
Trace[1755777139]: ---"Writing http response done" 37371ms (02:26:30.011)
Trace[1755777139]: [37.376677106s] [37.376677106s] END
```

### Kubernetes version

**k8s version** 
```
root@loshu-k8s-master01:/var/log# kubectl get nodes
NAME                   STATUS   ROLES           AGE   VERSION
loshu-k8s-master01     Ready    control-plane   30d   v1.26.6
loshu-k8s-master02     Ready    control-plane   30d   v1.26.6
loshu-k8s-master03     Ready    control-plane   30d   v1.26.6
```
**lb proxy** 
haproxy+keeaplive

### Cloud provider

on-premise

### OS version

ubuntu 22.04

### Install tools

kubeadm

### Container runtime (CRI) and version (if applicable)

**containerd**

### Related plugins (CNI, CSI, ...) and versions (if applicable)

**calico**
```[tasklist]
### Tasks
```

**Comment by neolit123** [comment]
/sig api-machinery
for triage

**Comment by MikeSpreitzer** [comment]
Yes, looks like a real bug, an oversight in logic.  Will fix.

**Comment by rmoreas** [comment]
Will this fix be backported to v1.26?

**Comment by prbakhsh** [comment]
@rmoreas I have k8s 1.26 and recently got same error.

**Comment by tmmorin** [comment]
I have observed the same error on v1.27.6

**Comment by kimabd** [comment]
I have observed the same error on v1.28.4

**Comment by n9** [comment]
Try to update to the latest patch version.

**Comment by Levivb** [comment]
@kimabd it seems fixed in `1.28.6`

See
https://github.com/kubernetes/kubernetes/blob/v1.28.6/staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go#L262

vs `1.28.5`
https://github.com/kubernetes/kubernetes/blob/v1.28.5/staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go#L262
